drone/src/agent/docker.rs

```diff
@@ -327,6 +331,15 @@ impl DockerInterface {
         Ok(ip.parse()?)
     }
 
+    /// returns true if image exists, returns false in all other cases
+    pub async fn image_exists(&self, image: &str) -> bool {
+        //not sure if it's faster to do this or check if image in docker.list_images
```

Comment:
This seems fine to me, and has the advantage of not being linear in the number of images.